stuffListApi/public/index.html

```diff
@@ -0,0 +1,127 @@
+
+<body>
+    <div class="container" ng-controller="StuffController">
+        <nav class="navbar navbar-default">
+            <div class="container-fluid">
+                <div class="navbar-header">
+                    <p class="navbar-brand" href="#">My Favorite Stuff</p>
+                </div>
+
+                <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
+                    <form class="navbar-form navbar-right">
+                        <div class="form-group">
+                            <input type="text" class="form-control" placeholder="Search stuffs" ng-disabled="!stuffs.length" ng-model="search" ng-model-options="{debounce: 600}">
+                            <button type="button" class="btn btn-primary" ng-click="createStuffDialog()">
+                                <span class="glyphicon glyphicon-plus"></span> Create Stuff
+                            </button>
+                            <button type="button" class="btn btn-success" ng-click="onFetchData()">
+                                <span class="glyphicon glyphicon-refresh"></span> Fetch/Reload
+                            </button>
+                            <button type="button" class="btn btn-danger" ng-disabled="!stuffs.length" ng-click="deleteAllStuffDialog()">
+                                <span class="glyphicon glyphicon-trash"></span> Delete All
+                            </button>
+                        </div>
+                    </form>
+                </div>
+                <!-- /.navbar-collapse -->
+            </div>
+            <!-- /.container-fluid -->
+        </nav>
+
+        <div id="spinner">
+            <img id="img-spinner" src="resources/spinner.gif" alt="Loading...">
+        </div>
+        <h3 ng-if="stuffs.length > 0">Stuffs which I love</h3>
+        <hr>
+        <div class="row" ng-if="stuffs.length > 0">
+            <div class="col-sm-6 col-md-4 fadeIn animated" ng-repeat="stuff in stuffs | filter:search">
+                <div class="thumbnail">
+                    <div class="caption">
+                        <h4>{{stuff.category}}</h4>
+                        <p>{{stuff.name}}</p>
+                        <button class="btn btn-default btn-xs" ng-click="editStuffDialog(stuff._id)">Edit</button>
+                        <button class="btn btn-danger btn-xs" ng-click="deleteStuffDialog(stuff._id)">Delete</button>
+                    </div>
+                </div>
+            </div>
+        </div>
+
+        <h3 ng-if="stuffs.length > 0">Categories</h3>
+        <hr>
+        <div class="row" ng-if="stuffs.length > 0">
+            <div class="col-sm-6 col-md-4 fadeIn animated" ng-repeat="stuff in stuffCategorized | filter:search">
+                <div class="thumbnail">
+                    <div class="caption">
+                        <h4>{{stuff.category}}</h4>
+                        <div>
+                            <p ng-repeat="stuffName in stuff.names">{{stuffName}}</p>
```

Comment:
Is this error coming because you've added same category and name? In this case maybe we can use something like this:

`ng-repeat="stuffName in stuff.names track by $index"`